examples/solvers/find_optimal_parameters.py

```diff
@@ -0,0 +1,189 @@
+"""
+Example of how to use optimization in order to pick reconstruction parameters.
+
+In this example, we solve the tomographic inversion problem with different
+regularizers (fbp, huber and tv) and pick the "best" regularization parameter
+for each method w.r.t. a set of reference data.
+
+To find the "best" parameter we use Powells method to optimize a figure of
+merit, here the L2-distance to the true result.
+"""
+
+import numpy as np
+import odl
+import scipy
+
+
+def optimal_parameters(reconstruction, fom, phantoms, data,
+                       initial_param=0):
+    """Find the optimal parameters for a reconstruction method.
+
+    Parameters
+    ----------
+    reconstruction : callable
+        Function that takes two parameters:
+
+            * data : The data to be reconstructed
+            * parameters : Parameters of the reconstruction method
+
+        The function should return the reconstructed image.
+    fom : callable
+        Function that takes two parameters:
+
+            * reconstructed_image
+            * true_image
+
+        and returns a scalar Figure of Merit.
+    phantoms : sequence
+        True images
+    data : sequence
+        The data to be reconstructed
+    initial_param : array-like
+        Initial guess for the parameters
+    """
+    def func(lam):
+        # Function to be minimized by scipy
+        return sum(fom(reconstruction(datai, lam), phantomi)
+                   for phantomi, datai in zip(phantoms, data))
+
+    # Pick resolution to fit the one used by the space
+    xtol = ftol = np.finfo(phantom.space.dtype).resolution * 10
+
```

Comment:
Here we could add an optimization in case the parameter is a scalar,

```python
    initial_param = np.asarray(initial_param)
    if initial_param.size == 1:
        bracket = [initial_param - tol, initial_param + tol]
        result = scipy.optimize.minimize_scalar(func,
                                                bracket=bracket,
                                                tol=tol,
                                                bounds=bounds,
                                                options={'disp': False})
        return result.x
    else:
        # Use a gradient free method to find the best parameters
        parameters = scipy.optimize.fmin_powell(func, initial_param,
                                                xtol=tol,
                                                ftol=tol,
                                                disp=False)
        return parameters
```